spec/unit/knife/bootstrap_windows_winrm_spec.rb

```diff
@@ -41,6 +42,109 @@
 
   let(:initial_fail_count) { 4 }
 
+  context "knife secret-file && knife secret options are passed" do
```

Comment:
I have modified the RSpecs to mock out the behavior instead of creating secret-file on the file system, keeping the specs under unit test cases.